lib/rules/amd-function-arity.js

```diff
@@ -30,10 +34,35 @@ module.exports = function (context) {
         return node.type === "FunctionExpression";
     }
 
+    function findAllowedExtraDependencies() {
+        var optionNotUndefined = context.options && context.options[0] && context.options[0].allowExtraDependencies;
+
+        if ((optionNotUndefined && context.options[0].allowExtraDependencies === true)) {
+            allowExtraDependencies = true;
+            allowedExtraDependencies = false;
+        } else if (optionNotUndefined) {
+            allowExtraDependencies = false;
+            allowedExtraDependencies = context.options[0].allowExtraDependencies;
+        } else {
+            allowExtraDependencies = false;
+            allowedExtraDependencies = false;
+        }
+    }
+
     function requireHasCallback(node) {
         return node.arguments.filter(isFunctionExpression).length;
     }
 
+    function getUnassignedPaths(dependencyNodes, callbackParams) {
+        return dependencyNodes.slice(callbackParams.length);
```

Comment:
This is close, but if i'm following correctly, this will grab the _first n_ dependency nodes, and what you want is the _last n_ dependency nodes. Subtracting `dependencyNodes.length` from `callbackParams.length` in my previous example gives us a negative number, which is what we want to pass to `slice` here, to slice from the end of the dependency node array.